ts/DataGrid/Table/Actions/RowsVirtualizer.ts

```diff
@@ -268,20 +268,38 @@ class RowsVirtualizer {
 
         const alwaysLastRow = rows.pop();
 
+        // Remove rows that are out of the range except the last row.
         for (let i = 0, iEnd = rows.length; i < iEnd; ++i) {
-            rows[i].destroy();
+            const row = rows[i];
+            const rowIndex = row.index;
+
+            if (rowIndex < from || rowIndex > to) {
+                !row.destroyed && row.destroy();
+            }
         }
-        rows.length = 0;
 
         for (let i = from; i <= to; ++i) {
-            const newRow = new TableRow(vp, i);
-            newRow.render();
-            vp.tbodyElement.insertBefore(
-                newRow.htmlElement,
-                vp.tbodyElement.lastChild
-            );
-
-            rows.push(newRow);
+            const row = rows[i];
+            let newRow: TableRow;
+
+            // Recreate row when it is destroyed and it is in the range.
+            if (row && row.destroyed) {
+                newRow = new TableRow(vp, i);
+                newRow.render();
+                vp.tbodyElement.insertBefore(
+                    newRow.htmlElement,
+                    vp.tbodyElement.lastChild
+                );
+                rows[i] = newRow;
+            } else if (!row) {
+                newRow = new TableRow(vp, i);
+                newRow.render();
+                vp.tbodyElement.insertBefore(
+                    newRow.htmlElement,
+                    vp.tbodyElement.lastChild
+                );
+                rows.push(newRow);
+            }
```

Comment:
In both (IF and ELSE) all lines (except the last one) are the same. So I think that we can optimize it.